developer-docs-site/docs/guides/building-your-own-wallet.md

```diff
@@ -88,9 +88,12 @@ There will be some apis that certain wallets may add but there should be a few a
 ```typescript
 // Common Args and Responses
 
+// For single-signer account, there is one publicKey and minKeysRequired is null.
+// For multi-signer account, there are multiple publicKeys and minKeysRequired value.
 interface PublicAccount {
-    string address;
-    string publicKey;
+    address: string;
+    publicKeys: string[];
+    minKeysRequired?: number; // for multi-signer account
```

Comment:
why does the dapp need to care about this? Is the dapp supposed to take into account the minKeysRequired when evaluating `signMessage`?

Reply:
for k-of-N multisig wallets, to verify a message validness dapp should know the `k` value, which we use `minKeysRequired` field for better readability, and for N dapp side can get from publicKeys.length https://aptos.dev/concepts/basics-accounts#multisigner-authentication

Reply:
Yes, when evaluating `signMessage`, the dApp needs at least `minKeysRequired` valid signatures.

Evaluate `signMessage` Step:
1. The number of `signatures` of `SignMessageResponse` is greater than `minKeysRequired`
2. Check every signature is valid by using `bitmap` to find corresponding to public key.